### DISCUSSION TITLE
could not connect to specific host from Linux server

### DISCUSSION THREAD

**Discussion by SuryaJD** [openssl/openssl#26072] [Q&A] [answered]
I have a Linux server and am trying to connect to a specific host, such as 'imap.ionos.com,' on a 993 port. I am using the php-imap package. But receivinga  connection timeout error To debug this more I tried to connect to the same host and port with telnet but could not connect. further debugging I could confirm that the error is with the OpenSSL.

here is the exact error.

> stream_socket_client(): Unable to connect to ssl://imap.ionos.com:993 (Connection timed out)

I am facing this same error while connecting via SMTP too.I am sure its not something related to IMAP or its package.

What's more frustrating is same code is working on staging and the local server. My first assumption was that IP had been blocked. I have assigned a different IP to the server and still no luck. I also checked the outbound firewall for any network restrictions.

OpenSSL is on the 3.0.1 version If that's a matter thanks.

![image](https://github.com/user-attachments/assets/d0f7f505-bfd9-4803-a14a-a2c5a2e11a60)

![image](https://github.com/user-attachments/assets/6d95ec5b-94ad-4e67-bc51-3c093a0fed45)

**Comment by nhorman** [comment]
whats the rest of the openssl s_client command with the trace option show?  It appears cut off in your image above

**Comment by mattcaswell** [comment, marked as answer]
> To debug this more I tried to connect to the same host and port with telnet but could not connect.

If you cannot connect with telnet then there is some fundamental network level problem preventing you from connecting, e.g. the port is blocked or similar.

Your s_client command indicates errno 110 which is "connection timed out" (consistent with the error message you posted in the description above). This confirms an underlying network issue. This isn't an OpenSSL issue. You need to figure out why you can't open that host and port.
